<launch>

  <group ns="calibration_config" clear_params="true">

    <group ns="sensors">
      <param name="all"    textfile="$(find pr2_se_calibration_launch)/estimate_params/config_pr2_se/sensors.yaml" />
    </group>

    <group ns="cal_steps">

      <group ns="SE - 00 - Estimating Checkerboard Locations">
        <param name="free_params" textfile="$(find pr2_se_calibration_launch)/estimate_params/config_pr2_se/free_cb_locations.yaml" />
        <param name="use_cov" type="bool" value="False" />
        <rosparam>
          sensors:
          - right_arm_chain
          - tilt_laser
        </rosparam>
        <param name="output_filename" type="string" value="head_laser_0" />
      </group>


      <group ns="SE - 01 - Adding Camera Locations">
        <param name="free_params" textfile="$(find pr2_se_calibration_launch)/estimate_params/config_pr2_se/free_cameras.yaml" />
        <param name="use_cov" type="bool" value="True" />
        <rosparam>
          sensors:
          - kinect_rect
          - prosilica_rect
          - tilt_laser
        </rosparam>
        <param name="output_filename" type="string" value="head_laser_1" />
      </group>

      <group ns="SE - 02 - Adding Laser Pose">
        <param name="free_params" textfile="$(find pr2_se_calibration_launch)/estimate_params/config_pr2_se/free_laser.yaml" />
        <param name="use_cov" type="bool" value="True" />
        <rosparam>
          sensors:
          - kinect_rect
          - prosilica_rect
          - tilt_laser
        </rosparam>
        <param name="output_filename" type="string" value="head_laser_3" />
      </group>

      <group ns="SE - 03 - Locking Head Sensors. Estimating Arm Joint Offsets">
        <param name="free_params" textfile="$(find pr2_se_calibration_launch)/estimate_params/config_pr2_se/free_arms.yaml" />
        <param name="use_cov" type="bool" value="True" />
        <rosparam>
          sensors:
          - kinect_rect
          - prosilica_rect
          - tilt_laser
          - right_arm_chain
          - forearm_right_rect
        </rosparam>
        <param name="output_filename" type="string" value="system_calibrated" />
      </group>

    </group>

    <param name="initial_system" textfile="$(find pr2_se_calibration_launch)/estimate_params/config_pr2_se/system.yaml" />

  </group>

</launch>
